catroid/src/main/java/org/catrobat/catroid/content/actions/HideTextAction.java

```diff
@@ -37,20 +37,22 @@
 public class HideTextAction extends TemporalAction {
 
 	private UserVariable variableToHide;
+
+	private Scene scene;
+	private Sprite sprite;
 	private UserBrick userBrick;
 
 	@Override
 	protected void begin() {
-		DataContainer dataContainer = ProjectManager.getInstance().getCurrentScene().getDataContainer();
-		Sprite currentSprite = ProjectManager.getInstance().getCurrentSprite();
+		DataContainer dataContainer = scene.getDataContainer();
```

Comment:
Actually we never need to pass the scene as a parameter, since there is always just one running scene. Action of Sprite is executed -> Sprite is part of running Scene -> Sprite is part of sceneToPlay. There should never ever be another possibility (only if at some point concurrent scene execution is introduced). This behaviour is defined [here](https://github.com/Catrobat/Catroid/blob/develop/catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java#L349).